colcon_cmake/task/cmake/build.py

```diff
@@ -148,25 +148,22 @@ async def _reconfigure(self, args, env):
         cmake_args += ['-DCMAKE_INSTALL_PREFIX=' + args.install_base]
         generator = get_generator(args.build_base, args.cmake_args)
         if os.name == 'nt' and generator is None:
-            vsv = get_visual_studio_version()
-            if vsv is None:
+            try:
+                msbuild_version = get_msbuild_version()
+            except FileNotFoundError as e:
                 raise RuntimeError(
-                    'VisualStudioVersion is not set, '
-                    'please run within a Visual Studio Command Prompt.')
-            supported_vsv = {
-                '16.0': 'Visual Studio 16 2019',
-                '15.0': 'Visual Studio 15 2017',
-                '14.0': 'Visual Studio 14 2015',
-            }
-            if vsv not in supported_vsv:
+                    'msbuild is not found, '
```

Comment:
This error message might be misleading in case the executable exists but fails for whatever reason. It would be good to distinguish these error cases to provide the correct information to the user.

Reply:
That error is only thrown on `FileNotFoundError`. Added some more explicit handling so the error message should always contain the entire version string.